OMCompiler/Compiler/NBackEnd/Modules/1_Main/NBPartitioning.mo

```diff
@@ -339,9 +339,9 @@ protected
     clone_eqns := EquationPointers.clone(equations);
     systems := {System.SYSTEM(
       systemType        = systemType,
-      unknowns          = if isInit then Initialization.sortInitVars(clone_vars) else clone_vars,
+      unknowns          = clone_vars,
       daeUnknowns       = NONE(),
-      equations         = if isInit then Initialization.sortInitEqns(clone_eqns) else clone_eqns,
+      equations         = clone_eqns,
```

Comment:
INIT doesn't need reordering anymore?

Reply:
No, i am doing it in multiple matching steps instead. It is safer than just ordering the variables and equations because i force the matching algorithm to respect the ordering. It also works with any matching algorithm and not only the ones actually working index first for both equations and variables.

Reply:
OK got it, perfect :ok_hand: